library/modules/Items.cpp

```diff
@@ -832,6 +831,26 @@ static bool detachItem(MapExtras::MapCache &mc, df::item *item)
         }
     }
 
+    if (df::general_ref *ref =
+                Items::getGeneralRef(item, general_ref_type::PROJECTILE))
+    {
+        int32_t proj_id = ((df::general_ref_projectile *)ref)->projectile_id;
+        df::proj_list_link *link = world->proj_list.next;
+        for (; link; link = link->next)
+        {
+            if (link->item->id != proj_id)
+                continue;
+
+            link->prev->next = link->next;
+            if (link->next)
+                link->next->prev = link->prev;
+            delete(link);
```

Comment:
I found `linked_list_append()` and `linked_list_insert_after()` in `MiscUtils.h` - maybe we could also use a `linked_list_remove()`?

Reply:
done.